tests/testthat/test-dataset-stream.R

```diff
@@ -0,0 +1,66 @@
+context("Dataset stream")
+# 
+mock_stream_rows <- data.frame(
+    birthyr = c(0.577530, 0.577530),
+    gender = c(2, 1),# Female , Male
+    mymrset = c(list(2, 1, 1)),
+    textVar = c("a", "b"),
+    starttime = c("1955-12-28", "1955-12-29")
+)
+
+with_mock_crunch({
+    ds <- loadDataset("test ds")   ## has 2 messages waiting, 4 rows received
+    ds2 <- loadDataset("an archived dataset", kind="archived") ## Has no streams mock
+    ds3 <- loadDataset("ECON.sav") ## has 0 messages waiting, 0 rows received
+    test_that("pendingStream gets pending messages", {
+        expect_equal(pendingStream(ds), 2)
+        expect_GET(pendingStream(ds2), 'https://app.crunch.io/api/datasets/2/stream')        
+    })
+    
+    test_that("streamRows streams rows", {
+        expect_equal(streamRows(ds, data=data.frame()), ds)
+        expect_POST(streamRows(ds, data=mock_stream_rows),
+                    'https://app.crunch.io/api/datasets/1/stream/',
+                    '{"birthyr":0.5775,"gender":2,"mymrset.2":2,"mymrset.1":1,"mymrset.1.1":1,',
```

Comment:
`"mymrset.2"` isn't right. Array data should be streamed keyed by alias of the subvariables (as if it wasn't an array). So for this mock, it's `subvar1`, `subvar2`, and `subvar3`. Edit your `mock_stream_rows` to replace `mymrset` with those.